cmd/filegen/main.go

```diff
@@ -451,6 +411,24 @@ func generateFiles(ctx *cli.Context) error {
 		if err != nil {
 			return err
 		}
+
+		numObservers := numOfShards*numOfObserversPerShard + numOfMetachainObservers
```

Comment:
You can extract this line together with the calculation below -> totalAddressesWithBalances-numObservers in one variable outside of this for and then you can use it directly in next line -> if i < precalculatedVariable. Actually you don't need to recalculate this in each iteration as this value would not be changed.

Reply:
done